api/pkg/apis/v1alpha1/managers/targets/targets-manager.go

```diff
@@ -27,6 +27,7 @@ type TargetsManager struct {
 	managers.Manager
 	StateProvider    states.IStateProvider
 	RegistryProvider registry.IRegistryProvider
+	NeedValidate     bool
```

Comment:
why are we setting up a manager level flag?

Reply:
State provider config is also manager-level config. We only skip validation if the manager uses k8s state provider - webhook will take over the validation in that case.